aws/aws-toolkit-vscode
# Fix JS CodeLenses to work with a wider variety of file layouts

## Description

The current Toolkit only shows JS CodeLenses when the handler code is in the same folder as the package.json (manifest) file. An example is shown in issue #757.

This change calculates the full relative path that would be expected on the JS Lambda Handler, and stores that in the CodeLens command. This allows the CodeLens command to find a match in SAM Templates, which then shows the missing CodeLenses.

## Testing





Tests added to verify that CodeLenses appear in different file layout scenarios.

Tested on Windows/Mac:
* Make a SAM app, placing the package.json file in the same location as the sam template.
* Place handler code two levels down ("/src/handlers") and craft SAM Template appropriately (Handler is src/handlers/app.myHandler)
* expect CodeLenses to appear on handler file
* expect CodeLenses to Run Local/Debug Local and configure properly


## Checklist




- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
